src/pages/index.js

```diff
@@ -47,61 +116,140 @@ cardAddFormValidator.enableValidation();
+      cardAddFormValidator.disableButton(); // Disable the form submission button
+      cardAddForm.reset();
+    })
+    .catch((error) => {
+      console.error("Error adding card:", error);
+    })
+    .finally(() => {
+      newCardModal.handleLoad(false); // Reset the button text
+    });
 }
 
 // Instance of PopupWithForm for adding a new card
 const newCardModal = new PopupWithForm("#card-add-modal", handleAddCardSubmit);
 newCardModal.setEventListeners();
 
-const addCardButton = document.querySelector('.profile__add-button');
+const addCardButton = document.querySelector(".profile__add-button");
 addCardButton.addEventListener("click", () => {
   newCardModal.open();
 });
 
-
-
-
 //----------------------------------------------------------------
 // Profile edit Modal
 
 //Profile Modal validation
-const editProfileFormValidator = new FormValidator(validationSettings, profileEditForm);
+const editProfileFormValidator = new FormValidator(
+  validationSettings,
+  profileEditForm
+);
 editProfileFormValidator.enableValidation();
 
 // Function to handle form submission and update the profile
 function handleProfileFormSubmit(input) {
-  userInfo.setUserInfo(input);
-  profileModal.close();
+  profileModal.handleLoad(true, "Saving...");
+  // userInfo.setUserInfo(input);
+  // profileModal.close();
+
+  api
+    .setUserInfo(input)
+    .then(() => {
+      userInfo.setUserInfo(input);
+      profileModal.close();
+      profileEditForm.reset();
+    })
+    .catch((error) => {
+      console.error("Error updating profile:", error);
+    })
+    .finally(() => {
+      profileModal.handleLoad(false); // Reset the button text
+    });
 }
 
 // create an instance of PopupWithForm for the profile modal
-const profileModal = new PopupWithForm( "#profile-edit-modal" , handleProfileFormSubmit);
+const profileModal = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileFormSubmit
+);
 profileModal.setEventListeners();
 
 // open the profile modal on button click
-const openProfileModalButton = document.querySelector('.profile__edit-button');
+const openProfileModalButton = document.querySelector(".profile__edit-button");
 openProfileModalButton.addEventListener("click", () => {
   const input = userInfo.getUserInfo();
   profileModal.setInputValues(input);
   editProfileFormValidator.resetValidation();
   profileModal.open();
 });
 
+// Card delete confirm Modal ------------------------------------------
+
+// const deleteConfirmationModal = new PopupWithConfirmation ('.confirm-modal', handleCardDelete );
+
+const confirmPopup = new PopupWithConfirmation(
+  "#modal-confirm",
+  (cardData, evt) => {
+    evt.preventDefault();
+    api
+      .deleteCard(cardData.cardId)
+      .then(() => {
+        cardData.cardEl.remove();
+        cardData.cardEl = null;
+        confirmPopup.close();
+      })
+      .catch((err) => alert(err))
+      .finally(() => confirmPopup.resetButtonText());
```

Comment:
Please, delete this code because you  don't have `resetButtonText` inside `PopupWithConfirmation`


It's not needed here actually

It should be just an empty function `() => {}`

```js
const confirmPopup = new PopupWithConfirmation("#modal-confirm", () => {});


```